### DISCUSSION TITLE
File Observer API for Plugin

### DISCUSSION THREAD

**Discussion by dikidjatar** [Acode-Foundation/Acode#1120] [Ideas]
I'm currently working on an Acode plugin, but I'm facing an issue: I need a way to detect real-time file changes in the currently open folder (like when a file is modified, added, deleted, etc.). Right now, I can only use FileEvents, but it only triggers when the file is active in the editor and doesn't work for files listed in the sidebar file list.

I wish there were an event listener for file changes, not just for the active file in the editor.

If Acode had a File Observer API that could provide events for any file changes, it would be incredibly helpful, eliminating the need to manually reload repeatedly.

**Comment by bajrangCoder** [comment]
This isn't real time(I guess).
https://github.com/Acode-Foundation/Acode/blob/df668621bc3cb606558021e0743d476bb7276789/src/plugins/sdcard/index.d.ts#L252

You can use it through `sdcard` global object.

**Reply by dikidjatar** [reply]
Thanks, I'll give it a try. But why isn't this in the plugin docs?

**Reply by bajrangCoder** [reply]
No idea, Actually there is no docs for any Cordova plugins
